Src/RE2.Managed/NativeMethods.cs

```diff
@@ -88,15 +88,36 @@ public static unsafe void GetNamedGroupsSetup(StringUtf8 pattern, int* numCaptur
             }
         }
 
-        public static unsafe bool MatchesNamedGroups(StringUtf8 pattern, StringUtf8 text, GroupNameHeader* groupNameHeaders, byte* groupNamesBuffer, Submatch* submatches)
+        public static unsafe void MatchesNamedGroups2(
+                StringUtf8 pattern,
+                StringUtf8 text,
+                GroupNameHeader** groupNameHeadersOut,
+                byte** groupNamesBufferOut,
+                int* numGroupNames,
+                Submatch*** matchesOut,
+                int* numMatchesOut,
+                int* numSubmatchesOut,
+                void** groupNameHeadersCleanupPtrOut,
+                void** groupNamesBufferCleanupPtrOut,
+                void** matchesCleanupPtrOut)
         {
             if (Environment.Is64BitProcess)
             {
-                return NativeMethodsX64.MatchesNamedGroups(pattern, text, groupNameHeaders, groupNamesBuffer, submatches);
+                NativeMethodsX64.MatchesNamedGroups2(pattern,
+                    text,
+                    groupNameHeadersOut,
+                    groupNamesBufferOut,
+                    numGroupNames,
+                    matchesOut,
+                    numMatchesOut,
+                    numSubmatchesOut,
+                    groupNameHeadersCleanupPtrOut,
+                    groupNamesBufferCleanupPtrOut,
+                    matchesCleanupPtrOut);
             }
             else
             {
-                return NativeMethodsX86.MatchesNamedGroups(pattern, text, groupNameHeaders, groupNamesBuffer, submatches);
+                throw new NotImplementedException();
```

Comment:
>throw new NotImplementedException(); [](http://example.com/codeflow?start=16&length=36)

TODO: add boilerplate for this

Reply:
why do we need this?
can't you call the nativemethodsx86...?

Reply:
@eddynaka 
I am having issues getting the unit tests to run for x86. I recommend we leave this unimplemented for now and push the x86 implementation and testing to a backlog item.

Reply:
I added x86 support, but it is untested.

I added a test task here: [https://github.com/microsoft/sarif-pattern-matcher/issues/386](https://github.com/microsoft/sarif-pattern-matcher/issues/386)